src/link-manager.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @fileoverview Link manager for controlling the link query.
+ */
+
+/** @const */
+genotet.linkManager = {};
+
+/**
+ * Object that stores the link views by their type.
+ */
+/** @type {!Object<!Array<{
+ *    target: genotet.View,
+ *    action: string,
+ *    response: string
+ *  }>>}
+ */
+genotet.linkManager.links = {};
+
+/** @const {!Array<string>} */
+genotet.linkManager.EXPRESSION_ACTIONS = ['addGeneProfile'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.NETWORK_ACTIONS = ['nodeClick', 'edgeClick'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.BINDING_ACTIONS = ['updateTrack', 'locus'];
+
+/**
+ * Initializes the link manager.
+ */
+genotet.linkManager.init = function() {
+  genotet.linkManager.links = {};
+};
+
+
+/**
+ * Set up the link manager.
+ */
+genotet.linkManager.link = function() {
+  console.log(genotet.linkManager.links);
+  for (var linkViewName in genotet.linkManager.links) {
+    var linkView = genotet.viewManager.views[linkViewName];
+    genotet.linkManager.links[linkViewName].forEach(function(object) {
+      $(linkView.getRenderer())
+        .on('genotet.' + object.action, function(event, component) {
+          switch (object.action) {
+            case 'nodeClick':
+              var node = component;
+              var genes = [node.id];
+              break;
+            case 'edgeClick':
+              var edge = component;
+              var genes = edge.id.split(',');
+              break;
+          }
+          var targetView = object.target;
+          switch (object.response) {
+            case 'addGeneProfile':
+              var matrixGeneNames = targetView.getData().matrix.geneNames;
+              genes.forEach(function(gene) {
+                var geneIndex = matrixGeneNames.indexOf(gene);
+                if (geneIndex != -1) {
+                  targetView.getPanel().signal(object.response, geneIndex);
```

Comment:
This is a bad idea. You should never signal on something that doesn't belong to this object. Especially in this case you are signaling are the other objects' children. This will make encapsulation meaningless.